src/actions/public/L2/feeActions.ts

```diff
@@ -0,0 +1,204 @@
+import { gasPriceOracleABI, gasPriceOracleAddress } from '@eth-optimism/contracts-ts'
+import { type Abi } from 'abitype'
+import {
+  Address,
+  BlockTag,
+  encodeFunctionData,
+  EncodeFunctionDataParameters,
+  EstimateGasParameters,
+  getContract,
+  PublicClient,
+  serializeTransaction,
+  TransactionSerializableEIP1559,
+  TransactionSerializedEIP1559,
+  Transport,
+} from 'viem'
+import { type Chain } from 'viem/chains'
+
+/**
+ * Bytes type representing a hex string with a 0x prefix
+ */
+type Bytes = `0x${string}`
+
+/**
+ * Options to query a specific block
+ */
+type BlockOptions = {
+  /**
+   * Block number to query from
+   */
+  blockNumber?: bigint
+  /**
+   * Block tag to query from
+   */
+  blockTag?: BlockTag
+}
+
+/**
+ * Options for all GasPriceOracle methods
+ */
+export type GasPriceOracleParameters = BlockOptions
+
+/**
+ * Options for specifying the transaction being estimated
+ */
+export type OracleTransactionParameters<
+  TAbi extends Abi | readonly unknown[],
+  TFunctionName extends string | undefined = undefined,
+> =
+  & EncodeFunctionDataParameters<TAbi, TFunctionName>
+  & Omit<TransactionSerializableEIP1559, 'data' | 'type'>
+  & GasPriceOracleParameters
+/**
+ * Options for specifying the transaction being estimated
+ */
+export type GasPriceOracleEstimator = <
+  TChain extends Chain | undefined,
+  TAbi extends Abi | readonly unknown[],
+  TFunctionName extends string | undefined = undefined,
+>(
+  client: PublicClient<Transport, TChain>,
+  options: OracleTransactionParameters<TAbi, TFunctionName>,
+) => Promise<bigint>
+
+/**
+ * Internal helper to serialize a transaction
+ */
+const transactionSerializer = <
```

Comment:
would prefer we keep to one function per file, helps keep testing clear and some of these things may be handy elsewhere!